File: packages/superwall/android/src/main/java/io/capawesome/capacitorjs/plugins/superwall/classes/events/CustomPaywallActionEvent.java
package io.capawesome.capacitorjs.plugins.superwall.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;

public class CustomPaywallActionEvent {

    @NonNull
    private final String name;

    public CustomPaywallActionEvent(@NonNull String name) {
        this.name = name;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject event = new JSObject();
        event.put("name", name);
        return event;
    }
}

File: packages/superwall/ios/Plugin/Classes/Events/CustomPaywallActionEvent.swift
import Foundation
import Capacitor

@objc public class CustomPaywallActionEvent: NSObject {
    private let name: String

    init(name: String) {
        self.name = name
    }

    public func toJSObject() -> JSObject {
        var event = JSObject()
        event["name"] = name
        return event
    }
}
